packages/mdl-menu/simple/foundation.js

```diff
@@ -0,0 +1,201 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {MDLFoundation} from 'mdl-base';
+import {cssClasses, strings, numbers} from './constants';
+import {clamp, bezierProgress} from '../util';
+
+export default class MDLSimpleMenuFoundation extends MDLFoundation {
+  static get cssClasses() {
+    return cssClasses;
+  }
+
+  static get strings() {
+    return strings;
+  }
+
+  static get numbers() {
+    return numbers;
+  }
+
+  static get defaultAdapter() {
+    return {
+      addClass: (/* className: string */) => {},
+      removeClass: (/* className: string */) => {},
+      hasClass: (/* className: string */) => {},
+      hasNecessaryDom: () => /* boolean */ false,
+      getInnerDimensions: () => /* { width: number, height: number } */ {},
+      setScale: (/* x: number, y: number */) => {},
+      setInnerScale: (/* x: number, y: number */) => {},
+      getItems: () => /* HTMLCollection */ {},
+      getItemYParams: (/* el: Element */) => {},
+      setTransitionDelay: (/* el: Element, value: string */) => {}
+    };
+  }
+
+  constructor(adapter) {
+    super(Object.assign(MDLSimpleMenuFoundation.defaultAdapter, adapter));
+  }
+
+  init() {
+    const {ROOT, OPEN} = MDLSimpleMenuFoundation.cssClasses;
+
+    if (!this.adapter_.hasClass(ROOT)) {
+      throw new Error(`${ROOT} class required in root element.`);
+    }
+
+    if (!this.adapter_.hasNecessaryDom()) {
+      throw new Error(`Required DOM nodes missing in ${ROOT} component.`);
+    }
+
+    if (this.adapter_.hasClass(OPEN)) {
+      this.isOpen_ = true;
+    } else {
+      this.isOpen_ = false;
+    }
+
+    this.startScaleX_ = 0;
+    this.startScaleY_ = 0;
+    this.targetScale_ = 1;
+    this.scaleX_ = 0;
+    this.scaleY_ = 0;
+    this.running_ = false;
+  }
+
+  // Calculate transition delays for individual menu items, so that they fade in one at a time.
+  applyTransitionDelays_() {
+    const items = this.adapter_.getItems();
+    const dimensions = this.dimensions_;
+    const transitionDuration = MDLSimpleMenuFoundation.numbers.TRANSITION_DURATION_MS / 1000;
+    const start = MDLSimpleMenuFoundation.numbers.TRANSITION_SCALE_ADJUSTMENT_Y;
+
+    if (items) {
+      for (let i = 0; i < items.length; i++) {
+        const item = items[i];
+        const itemYParams = this.adapter_.getItemYParams(item);
+        this.itemHeight_ = itemYParams.height;
+        let itemDelayFraction = null;
+        if (this.adapter_.hasClass(MDLSimpleMenuFoundation.cssClasses.BOTTOM_LEFT) ||
+            this.adapter_.hasClass(MDLSimpleMenuFoundation.cssClasses.BOTTOM_RIGHT)) {
+          itemDelayFraction = ((dimensions.height - itemYParams.top - itemYParams.height) / dimensions.height);
+        } else {
+          itemDelayFraction = ((itemYParams.top / dimensions.height));
+        }
+        const itemDelay = (start + itemDelayFraction * (1 - start)) * transitionDuration;
```

Comment:
do you think we should call `toFixed(3)` on this so that it's normalized across CSS engines? https://www.w3.org/TR/css3-values/#numeric-types